src/seccompiler/src/lib.rs

```diff
@@ -110,32 +127,97 @@ pub fn compile_bpf(
                         // For `ioctls` we need to mask upper bits as musl
                         // sets them to 1, but libseccomp explicitly checks that they are 0.
                         // with 0x00000000FFFFFFFF mask upper bits are always 0.
-                        let op = if syscall == IOCTL {
-                            let original_rule = rule.op.to_scmp_type();
-                            if original_rule == ScmpCompareOp::Equal {
-                                ScmpCompareOp::MaskedEqual(0x00000000FFFFFFFF)
-                            } else {
-                                original_rule
+                        match rule.op {
+                            SeccompCmpOp::Eq => {
+                                if syscall == IOCTL {
+                                    scmp_arg_cmp {
+                                        arg: rule.index as u32,
+                                        op: scmp_compare::SCMP_CMP_MASKED_EQ,
+                                        datum_a: 0x00000000FFFFFFFF,
+                                        datum_b: rule.val,
+                                    }
+                                } else {
+                                    scmp_arg_cmp {
+                                        arg: rule.index as u32,
+                                        op: scmp_compare::SCMP_CMP_EQ,
+                                        datum_a: rule.val,
+                                        datum_b: 0,
+                                    }
+                                }
                             }
-                        } else {
-                            rule.op.to_scmp_type()
-                        };
-                        ScmpArgCompare::new(rule.index as u32, op, rule.val)
+                            SeccompCmpOp::Ge => scmp_arg_cmp {
+                                arg: rule.index as u32,
+                                op: scmp_compare::SCMP_CMP_GE,
+                                datum_a: rule.val,
+                                datum_b: 0,
+                            },
```

Comment:
nit: this construction can probably go into a utility function? no strong opinions though

Reply:
done